src/platforms/dialogflow/DialogFlowEvent.ts

```diff
@@ -84,14 +85,19 @@ export class DialogFlowEvent extends VoxaEvent {
    * After that we'll default to the userStorage value
    */
   protected initUser(): void {
+    const { originalDetectIntentRequest } = this.rawEvent;
     const { conv } = this.google;
     const storage = conv.user.storage as any;
     let userId: string;
 
+    this.source = _.get(originalDetectIntentRequest, "source", "google");
+
     if (_.get(storage, "voxa.userId")) {
       userId = storage.voxa.userId;
     } else if (conv.user.id) {
       userId = conv.user.id;
+    } else if (this.source === "facebook") {
+      userId = _.get(originalDetectIntentRequest, "payload.data.sender.id", v1());
     } else {
       userId = v1();
```

Comment:
now that i think about it, is this even correct? seems like it will generate a new userId on every request, is not stable

Reply:
That's what I was getting when I was testing, but not sure how much would affect to google assistant if I removed this, what do you think?

Reply:
i don't think it will affect google assistant as there's code in place to make sure we always get a userId if needed